profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt

```diff
@@ -1,44 +1,69 @@
 package com.imashnake.animite.profile
 
 import androidx.compose.foundation.background
-import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.res.stringResource
-import com.imashnake.animite.core.R as coreR
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.imashnake.animite.core.ui.LocalPaddings
-import com.imashnake.animite.core.ui.ProgressIndicator
+import com.imashnake.animite.profile.dev.internal.ANILIST_AUTH_DEEPLINK
+import com.imashnake.animite.profile.dev.internal.ANILIST_AUTH_URL
+import com.ramcosta.composedestinations.annotation.DeepLink
 import com.ramcosta.composedestinations.annotation.Destination
 import com.ramcosta.composedestinations.annotation.RootNavGraph
 
-@Destination(route = "profile-screen")
+@Destination(
+    route = "user",
+    deepLinks = [
+        DeepLink(
+            uriPattern = ANILIST_AUTH_DEEPLINK
+        )
+    ]
+)
 @RootNavGraph(start = true)
 @Composable
-fun ProfileScreen() {
+fun ProfileScreen(
+    viewModel: ProfileViewModel = hiltViewModel(),
+    accessToken: String? = null
+) {
+    accessToken?.let { viewModel.setAccessToken(it) }
+    val actualAccessToken by viewModel.accessToken.collectAsState()
+    val viewer by viewModel.viewer.collectAsState()
+
     Box(
         contentAlignment = Alignment.Center,
         modifier = Modifier
             .fillMaxSize()
             .background(MaterialTheme.colorScheme.background)
     ) {
-        Column(
-            horizontalAlignment = Alignment.CenterHorizontally,
-            verticalArrangement = Arrangement.spacedBy(
-                LocalPaddings.current.tiny
-            )
-        ) {
-            Text(
-                text = stringResource(coreR.string.coming_soon),
-                color = MaterialTheme.colorScheme.onSurfaceVariant,
-                style = MaterialTheme.typography.labelLarge
-            )
-            ProgressIndicator()
+        val uriHandler = LocalUriHandler.current
+
+        if (actualAccessToken == null) {
+            Button(onClick = { uriHandler.openUri(ANILIST_AUTH_URL) }) {
+                Text(text = "Log in")
+            }
+        } else {
+            viewer.data?.let { viewer ->
+                with(viewer) {
+                    Column(Modifier.padding(LocalPaddings.current.large)) {
```

Comment:
It might be a good idea to show a "Profile screen coming soon" or something, for the 3 people that use this outside of a dev context

Reply:
💀